.github/workflows/link-check.yml

```diff
@@ -0,0 +1,22 @@
+name: check Links
+
+on:
+  push:
+    paths:
+      - 'pages/blog/posts/**/*.md'
+
+  pull_request:
+    paths:
+      - 'pages/blog/posts/**/*.md' 
```

Comment:
I don't think we want to limit the link checker to only work for the blog. Should be for the whole site.

Reply:
Here, I would suggest to run this workflow only in case of pushes to the main branch and manual workflow dispatch and not in case of pull requests because of the way lychee will check our relative links i.e. lychee will reach out to the deployed website and attempt to resolve our relative links from the root page of the website which will often fail because the PR changes wouldn't be reflected on the deployment. 

Note: When using relative links in our blogs we will have to use root paths `/blog/posts/hello-world-hello-postman` instead of `./hello-world-hello-postman`.

Reply:
No, we DO want it to be on pull request. We want to avoid deploying with a broken link.
We need to work out how to also check the deployable version in the PR.
It might be a case of running the build before link checking, then doing a remap as part of the config to address localhost.

Reply:
Absolutely, thank you for your patience on this. I'll work on a solution to check the deployable version.